docs/v0.4/getting-started/docker-for-mac.md

```diff
@@ -160,34 +160,101 @@ NAME     LATEST IMAGE
 square   index.docker.io/$DOCKER_ID/square@sha256:ac089ca183368aa831597f94a2dbb462a157ccf7bbe0f3868294e15a24308f68   square.js   <empty>   <empty>   Ready    1m13s
 ```
 
-## create a deployer
+## create a Knative deployer
+
+The Knative runtime is only available on clusters with Istio and Knative installed. Knative deployers run riff workloads using Knative resources which provide auto-scaling (including scale-to-zero) based on HTTP request traffic, and routing.
+
+```sh
+riff knative deployer create knative-square --function-ref square --tail
+```
+
+After the deployer is created, you can see the hostname by listing deployers.
+
+```sh
+riff knative deployer list
+```
+```
+NAME             TYPE       REF      HOST                                 STATUS   AGE
+knative-square   function   square   knative-square.default.example.com   Ready    28s
+```
+
+### invoke the function
+
+Knative configures HTTP routes on the istio-ingressgateway. Requests are routed by hostname.
+
+Look up the nodePort for the ingressgateway. You should a port value like 30195:
+
+```sh
+INGRESS_PORT=$(kubectl get svc istio-ingressgateway --namespace istio-system --output 'jsonpath={.spec.ports[?(@.port==80)].nodePort}')
+echo $INGRESS_PORT
+```
+
+Invoke the function by POSTing to the ingressgateway, passing the hostname and content-type as headers:
 
-Deployers take a built function and deploy it to a riff runtime. There are two runtimes: core and knative. The core runtime is always available, while the knatve runtime is only available on clusters with Istio and Knative installed.
+```sh
+curl http://localhost:$INGRESS_PORT/ -w '\n' \
```

Comment:
How is this port exposed on localhost?

Reply:
it's on a nodePort since we installed Istio with` --set gateways.istio-ingressgateway.type=NodePort`